airbyte-scheduler/src/main/java/io/airbyte/scheduler/persistence/DefaultJobCreator.java

```diff
@@ -140,6 +142,28 @@ public long createSyncJob(SourceConnection source,
         destinationDockerImageName);
   }
 
+  // Strategy:
+  // 1. Set all streams to full refresh.
+  // 2. Create a job where the source emits no records.
+  // 3. Run a sync from the empty source to the destination. This will overwrite all data for each
+  // stream in the destination.
+  // 4. The Empty source emits no state message, so state will start at null (i.e. start from the
+  // beginning on the next sync).
+  @Override
+  public long createResetConnectionJob(DestinationConnection destination, StandardSync standardSync, String destinationDockerImage)
+      throws IOException {
+    final ConfiguredAirbyteCatalog configuredAirbyteCatalog = AirbyteProtocolConverters.toConfiguredCatalog(standardSync.getSchema());
```

Comment:
This covers `User case 1` but I don't think this really accomplishes `User case 2: Get a blank slate in my destination.` since it only covers currently configured streams.

I think this only solves `User case 1`, but that's the main problem to solve anyways. `User case 2` doesn't seem as important and beyond the scope of this PR.